browser-switch/src/main/java/com/braintreepayments/api/BrowserSwitchClient.java

```diff
@@ -116,17 +116,20 @@ public BrowserSwitchResult deliverResult(@NonNull FragmentActivity activity) {
             return null;
         }
 
-        BrowserSwitchResult result;
+        BrowserSwitchResult result = null;
 
         Uri deepLinkUrl = intent.getData();
         if (deepLinkUrl != null && request.matchesDeepLinkUrlScheme(deepLinkUrl)) {
             result = new BrowserSwitchResult(BrowserSwitchStatus.SUCCESS, request, deepLinkUrl);
-        } else {
+            persistentStore.clearActiveRequest(appContext);
+        } else if (request.getShouldNotifyCancellation()) {
+            // ensure that cancellation result is delivered exactly one time
             result = new BrowserSwitchResult(BrowserSwitchStatus.CANCELED, request);
+
+            request.setShouldNotifyCancellation(false);
```

Comment:
If `deliverResult` gets called again and the request `shouldNotifyCancellation` has been set to false already, how does that request ever get cleared out of the `persistentSore` now?

Reply:
It'll lay dormant until a success flow is triggered. Also, a new request will overwrite a pending one since there can only be one pending browser switch request at a time.